plugins/sequence/src/DivSequenceRenderer/components/DivSequenceRendering.tsx

```diff
@@ -93,27 +92,27 @@ function SequenceDivs({ features, regions, bpPerPx }: MyProps) {
         }
         switch (letter.toLocaleLowerCase()) {
```

Comment:
what about

```
diff --git a/plugins/sequence/src/DivSequenceRenderer/components/DivSequenceRendering.tsx b/plugins/>
index 3ae936fef..0eb29fee7 100644
--- a/plugins/sequence/src/DivSequenceRenderer/components/DivSequenceRendering.tsx
+++ b/plugins/sequence/src/DivSequenceRenderer/components/DivSequenceRendering.tsx
@@ -90,33 +90,11 @@ function SequenceDivs({ features, regions, bpPerPx }: MyProps) {
           style.boxSizing = 'border-box'
           style.borderCollapse = 'collapse'
         }
-        switch (letter.toLocaleLowerCase()) {
-          case 'a': {
-            const baseColor = theme.palette.bases.A.main
-            style.backgroundColor = baseColor
-            style.color = theme.palette.bases.A.contrastText
-            break
-          }
-          case 'g': {
-            const baseColor = theme.palette.bases.G.main
-            style.backgroundColor = baseColor
-            style.color = theme.palette.bases.G.contrastText
-            break
-          }
-          case 'c': {
-            const baseColor = theme.palette.bases.C.main
-            style.backgroundColor = baseColor
-            style.color = theme.palette.bases.C.contrastText
-            break
-          }
-          case 't': {
-            const baseColor = theme.palette.bases.T.main
-            style.backgroundColor = baseColor
-            style.color = theme.palette.bases.T.contrastText
-            break
-          }
-          default:
-            break
+        // @ts-ignore
+        const color = theme.palette.bases[letter.toUpperCase()]
+        if (color) {
+          style.backgroundColor = color.main
+          style.color = color.contrastText
         }

```

ts-ignore isn't that nice but it is more agnostic to lettering

Reply:
Good idea, I added that along with a way to work around the ts-ignore

```diff
diff --git a/plugins/sequence/src/DivSequenceRenderer/components/DivSequenceRendering.tsx b/plugins/sequence/src/DivSequenceRenderer/components/DivSequenceRendering.tsx
--- a/plugins/sequence/src/DivSequenceRenderer/components/DivSequenceRendering.tsx
+++ b/plugins/sequence/src/DivSequenceRenderer/components/DivSequenceRendering.tsx
@@ -90,9 +90,13 @@ function SequenceDivs({ features, regions, bpPerPx }: MyProps) {
           style.boxSizing = 'border-box'
           style.borderCollapse = 'collapse'
         }
-        // @ts-ignore
-        const color = theme.palette.bases[letter.toUpperCase()]
-        if (color) {
+        type Base = keyof typeof theme.palette.bases
+        function isBase(l: string): l is Base {
+          return Object.keys(theme.palette.bases).includes(l as Base)
+        }
+        const upperLetter = letter.toUpperCase()
+        if (isBase(upperLetter)) {
+          const color = theme.palette.bases[upperLetter]
           style.backgroundColor = color.main
           style.color = color.contrastText
```